src/main/scala/org/bblfsh/client/v2/BblfshClient.scala

```diff
@@ -0,0 +1,161 @@
+import io.grpc.ManagedChannelBuilder
+import gopkg.in.bblfsh.sdk.v2.protocol.driver.{ParseRequest, ParseResponse, DriverGrpc, SupportedLanguagesRequest, SupportedLanguagesResponse, VersionRequest, VersionResponse, DriverHostGrpc}
+
+import org.bblfsh.client.v2.libuast.Libuast
+
+
+class BblfshClient(host: String, port: Int, maxMsgSize: Int) {
+  private val DEFAULT_TIMEOUT_SEC = 5
+
+  private val channel = ManagedChannelBuilder
+    .forAddress(host, port)
+    .usePlaintext(true)
+    .maxInboundMessageSize(maxMsgSize)
+    .build()
+
+  private val stub = DriverGrpc.blockingStub(channel)
+  private val stubInfo = DriverHostGrpc.blockingStub(channel)
+
+  /**
+    * Parses file with a given name and content using 
+    * the provided timeout.
+    *
+    * @param name file name
+    * @param content file content
+    * @param lang (optional) language to parse, default auto-detect \w enry
+    * @param timeout (disabled) bblfsh request timeout, seconds
+    *                Right now this does not have any effect in v2.
+    * @return UAST in parse response.
+    */
+  def parseWithOptions(
+    name: String,
+    content: String,
+    lang: String,
+    timeout: Long
+  ): ParseResponse = {
+    // FIXME(bzz): make timout work in v2 again
+    val req = ParseRequest(
+      filename = name,
+      content = content,
+      language = lang
+    )
+    stub.parse(req)
+  }
+
+  /**
+    * Parses the given file name and content.
+    *
+    * A default timeout of 5 seconds will be applied, same as was done by the server
+    * before https://github.com/bblfsh/bblfshd/blob/83166ea0087bfe20c24fc471309f70f422383198/daemon/pool.go#L191
+    *
+    * @param name file name
+    * @param content file content
+    * @param lang (optional) language to parse, default auto-detect \w enry
+    * @return UAST in parse response.
+    */
+  def parse(
+    name: String,
+    content: String,
+    lang: String = ""
+  ): ParseResponse = parseWithOptions(name, content, lang, DEFAULT_TIMEOUT_SEC)
+
+  /**
+    * Parses the given file name and content,
+    * setting a bblfsh request timeout.
+    *
+    * Since v1.11, this API exposes the timeout.
+    *
+    * @param name file name
+    * @param content file content
+    * @param timeout bblfsh request timeout, seconds
+    * @param lang (optional) language to parse, default auto-detect \w enry
+    * @return
+    */
+  def parseWithTimeout(
+    name: String,
+    content: String,
+    timeout: Long,
+    lang: String = ""
+  ): ParseResponse = parseWithOptions(name, content, lang, timeout)
+
+  def supportedLanguages(): SupportedLanguagesResponse = {
+    val req = SupportedLanguagesRequest()
+    stubInfo.supportedLanguages(req)
+  }
+
+  def version(): VersionResponse = {
+    val req = VersionRequest()
+    stubInfo.serverVersion(req)
+  }
+
+  def close(): Unit = {
+    channel.shutdownNow()
+  }
+}
+
+object BblfshClient {
+  val DEFAULT_MAX_MSG_SIZE = 100 * 1024 * 1024 // bytes
+
+  val PreOrder      = 0
```

Comment:
These don't seem to be used anywhere—are they mean to support the iterator implementation that is coming later?

Reply:
Yes, exactly - this is part of the old code that was moved forward and going to be used for iterator implementation.